test/adapter/json_api/parse_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          class ParseTest < Minitest::Test
+            def setup
```

Comment:
>  introducing a dependency to the serialization would add

I mean like

1. Given: `{ 'data' : { 'posts' : [ { 'id' : '1' } ] } }`
1. Deserialize to `[ Post.new(id: 1) ]`
1. Serialize to `{ 'data' : { 'posts' : [ { 'id' : '1' } ] } }`